### DISCUSSION TITLE
posts/why-ienumerable-no-foreach/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#8] [Announcements]
# posts/why-ienumerable-no-foreach/

为什么 IEnumerable 不像 List 那样拥有一个 ForEach 方法？

https://blog.coldwind.top/posts/why-ienumerable-no-foreach/

**Comment by MUSIC-DIE** [comment]
经常想ForEach  Ienumable   但是不清楚为什么不能这样做！

**Comment by staoran** [comment]
年轻时喜欢用 ForEach ，现在都是用 foreach 。😄

**Reply by BYJRK** [reply]
最新版本的 .NET 对 `foreach` 又有优化，现在效率和 `for` 基本上不相上下了

**Comment by Nitload-NSI** [comment]
| ForEach | 1,785.0 us |       88 B |
这个地方这么慢是不是跟其使用委托间接调用逻辑，难以进行充分地内联展开有关啊
